### PR TITLE
Improve refreshCampaigns endpoint to import and update campaigns automatically.

### PR DESCRIPTION
**Pull request by LATaylor-guardian**

### CODE REVIEW

File: app/repositories/contentapi/CapiSectionTransformer.scala [hunk 1]
```diff
@@ -0,0 +1,36 @@
...
+    section.activeSponsorships.flatMap(_.headOption.map(_.sponsorshipType.name))
+
+  def deriveSponsorshipLogo(section: CapiSection): Option[String] =
+    section.activeSponsorships.getOrElse(Nil).headOption.map(sponsorship => sponsorship.sponsorLogo)
+
+  def deriveStatus(startDate: Option[DateTime], endDate: DateTime): String = {
+    (startDate, endDate) match {
+      case (_, ed) if ed.isBeforeNow => "dead"
+      case (Some(sd), _)             => "live"
+      case _                         => "unknown"
```
Comment (kelvin-chappell) [inline app/repositories/contentapi/CapiSectionTransformer.scala]: In this case there's no start date and end date is in future, which would make it `live`.

File: app/services/CampaignService.scala [hunk 2]
```diff
@@ -97,4 +94,103 @@ object CampaignService {
...
+    val contentItems: List[ContentItem] = CapiContentTransformer.buildContentItems(apiContent, campaign.id)
+
+    val putContentResults: List[Either[CampaignCentralApiError, PutContentItemResult]] = for {
+      item <- contentItems
+    } yield CampaignContentRepository.putContent(item)
+
+    val results: (List[CampaignCentralApiError], List[PutContentItemResult]) =
+      putContentResults.foldRight((List[CampaignCentralApiError](), List[PutContentItemResult]())) {
+        case (itemResultOrError, (ls, rs)) => itemResultOrError.fold(l => (l :: ls, rs), r => (ls, r :: rs))
+      }
```
Comment (kelvin-chappell) [inline app/services/CampaignService.scala]: If only interested in first error, could do:
```
putContentResults.collectFirst {
      case Left(e) => e
    } toLeft {
      putContentResults collect {
        case Right(result) => result
      }
    }
```

**Review by kelvin-chappell — APPROVED** [review]
Looks good! 👍 

Not sure what triggers the refresh.

**Comment by LATaylor-guardian** [comment]
@kelvin-chappell 
>Not sure what triggers the refresh.

I believe the lambda swells wrote does: https://github.com/guardian/campaign-central-schedule-lambdas/blob/master/functions/expireCampaigns/main.go#L64

File: app/services/CampaignService.scala (hunk 2 above)
Comment (LATaylor-guardian) [reply app/services/CampaignService.scala]: Thanks, I just moved this code from another file but will change as much clearer.

File: app/repositories/contentapi/CapiSectionTransformer.scala (hunk 1 above)
Comment (LATaylor-guardian) [reply app/repositories/contentapi/CapiSectionTransformer.scala]: Thanks will update.

**Pull request merged**